cyclonedx/model/component.py

```diff
@@ -26,9 +26,181 @@
+
+    .. note::
+        See the CycloneDX Schema definition: https://cyclonedx.org/docs/1.4/xml/#type_commitType
+    """
+
+    def __init__(self, uid: Optional[str] = None, url: Optional[XsUri] = None,
+                 author: Optional[IdentifiableAction] = None, committer: Optional[IdentifiableAction] = None,
+                 message: Optional[str] = None) -> None:
+        if not uid and not url and not author and not committer and not message:
+            raise NoPropertiesProvidedException(
+                'At least one of `uid`, `url`, `author`, `committer` or `message` must be provided for a `Commit`.'
+            )
+
+        self.uid = uid
+        self.url = url
+        self.author = author
+        self.committer = committer
+        self.message = message
+
+    @property
+    def uid(self) -> Optional[str]:
+        """
+        A unique identifier of the commit. This may be version control specific. For example, Subversion uses revision
+        numbers whereas git uses commit hashes.
+
+        Returns:
+            `str` if set else `None`
+        """
+        return self._uid
+
+    @uid.setter
+    def uid(self, uid: Optional[str]) -> None:
+        self._uid = uid
+
+    @property
+    def url(self) -> Optional[XsUri]:
+        """
+        The URL to the commit. This URL will typically point to a commit in a version control system.
+
+        Returns:
+             `XsUri` if set else `None`
+        """
+        return self._url
+
+    @url.setter
+    def url(self, url: Optional[XsUri]) -> None:
+        self._url = url
+
+    @property
+    def author(self) -> Optional[IdentifiableAction]:
+        """
+        The author who created the changes in the commit.
+
+        Returns:
+            `IdentifiableAction` if set else `None`
+        """
+        return self._author
+
+    @author.setter
+    def author(self, author: Optional[IdentifiableAction]) -> None:
+        self._author = author
+
+    @property
+    def committer(self) -> Optional[IdentifiableAction]:
+        """
+        The person who committed or pushed the commit
+
+        Returns:
+            `IdentifiableAction` if set else `None`
+        """
+        return self._committer
+
+    @committer.setter
+    def committer(self, committer: Optional[IdentifiableAction]) -> None:
+        self._committer = committer
+
+    @property
+    def message(self) -> Optional[str]:
+        """
+        The text description of the contents of the commit.
+
+        Returns:
+            `str` if set else `None`
+        """
+        return self._message
+
+    @message.setter
+    def message(self, message: Optional[str]) -> None:
+        self._message = message
+
+    def __eq__(self, other: object) -> bool:
+        if isinstance(other, Commit):
+            return hash(other) == hash(self)
+        return False
+
+    def __hash__(self) -> int:
+        return hash((self.uid, hash(self.url), hash(self.author), hash(self.committer), self.message))
```

Comment:
no need to call `hash()` on each element in the tuple constructor.
the tuple's implementation of `hash` is expected to call `hash` in the tuples elements individually, if needed.
this means: just drop in all elements ala `hash((self.uid, self.url, self.author, self.committer, self.message))`

sample code for testing:
```py
class C:
    def __init__(self,a,b):
        self.a = a 
        self.b = b 

    def __hash__(self):
        h = hash((self.a, self.b))
        print( 'hash of ', id(self), repr(self), repr(self.a), repr(self.b), 'is', h)
        return h

i = C(C('foo', None),'bar')

print(hash(i))
```